docs/proposals/client-settings.md

```diff
@@ -0,0 +1,395 @@
+In this scenario, the HTTPRoutes share "ownership" of the server block and the external location block.
+
+### Creating the Effective Policy in NGINX Config
+
+To determine how to reliably and consistently create the effective policy in NGINX config, we need to apply the policies for each attachment scenario to the three NGINX mappings.
+
+**1. Gateway Attachment**
+
+A. Distinct Hostname:
+![example-a1](/docs/images/client-settings/example-a1.png)
+
+B. Same Hostname:
+![example-b1](/docs/images/client-settings/example-b1.png)
+
+C. Internal Redirect:
+![example-c1](/docs/images/client-settings/example-c1.png)
+
+For this attachment scenario, specifying the directives in every server context creates the effective policies for the attached HTTPRoutes. Specifying the directives in the http context would have the same effect, but this would not work once we add support for [multiple Gateway resources](https://github.com/nginxinc/nginx-gateway-fabric/issues/1443).
+
+**2. HTTPRoute Attachment**
+
+A. Distinct Hostname:
+![example-a2](/docs/images/client-settings/example-a2.png)
+
+B. Same Hostname:
+![example-b2](/docs/images/client-settings/example-b2.png)
+
+C. Internal Redirect
+![example-c2](/docs/images/client-settings/example-c2.png)
+
+For this attachment scenario, specifying the directives in the _final_ location blocks generated from the HTTPRoute with the policy attached achieves the effective policy. _Final_ means the location that ultimately handles the request.
+
+**3. Gateway and HTTPRoute Attachment**
+
+A. Distinct Hostname:
+![example-a3](/docs/images/client-settings/example-a3.png)
+
+B. Same Hostname:
+![example-b3](/docs/images/client-settings/example-b3.png)
+
+C. Internal Redirect
+![example-c3](/docs/images/client-settings/example-c3.png)
+
+For this attachment scenario, specifying the directives in the server and the _final_ location blocks creates the effective policies. All settings specified in the Gateway's policy are set as directives in the server block. For the HTTPRoute policies, only the settings whose values differ from the Gateway policy's settings are set in the location contexts.
+
+#### Overall Strategy
+
+The findings in the prior section can be condensed and generalized into the following two rules:
+
+- When a `ClientSettingsPolicy` is attached to a Gateway, add the corresponding NGINX directives to each server block generated from that Gateway.
+- When a `ClientSettingsPolicy` is attached to an HTTPRoute, compute the set of values that differ from the accepted `ClientSettingsPolicy` attached to its Gateway. For each value in this set, add the corresponding NGINX directive to each of the _final_ location blocks generated for the HTTPRoute. If no accepted `ClientSettingsPolicy` is attached to its Gateway, then the set of values contains all the values in the HTTPRoute's policy.
```

Comment:
> compute the set of values that differ from the accepted `ClientSettingsPolicy` attached to its Gateway

why not put all values in location context and let NGINX compute the effect set

Reply:
Good catch, we don't need to compute the set. I'll update